pageserver/src/tenant/timeline.rs

```diff
@@ -3221,7 +3221,7 @@ impl Timeline {
             // We don't return a blanket [`GetVectoredError::MissingKey`] to avoid
             // stalling compaction.
             keyspace.remove_overlapping_with(&KeySpace {
-                ranges: vec![NON_INHERITED_RANGE, NON_INHERITED_SPARSE_RANGE],
+                ranges: vec![NON_INHERITED_RANGE, Key::sparse_non_inherited_keyspace()],
```

Comment:
nit: could use a constant here instead

Reply:
I've updated the function to be a const fn. The reason why it's a function is that in the future it is possible that non_inherited range is not a single key range, but several keyspaces.